### PR TITLE
start null safety migration

### PR DESCRIPTION
**Pull request by jaumard**

### CODE REVIEW

File: lib/flutter_camera_ml_vision.dart [hunk 1]
```diff
@@ -152,18 +145,20 @@ class CameraMlVisionState<T> extends State<CameraMlVision<T>>
...
   ImageRotation get imageRotation => _rotation;
 
   Future<void> Function() get prepareForVideoRecording =>
       _cameraController.prepareForVideoRecording;
 
-  Future<void> startVideoRecording(String path) async {
+  Future<void> startVideoRecording() async {
     await _cameraController.stopImageStream();
-    return _cameraController.startVideoRecording(path);
+    return _cameraController.startVideoRecording();
```
Comment (kevin-sakemaer) [inline lib/flutter_camera_ml_vision.dart]: The function return type is Future<void> look like the is a type mismatch

File: lib/flutter_camera_ml_vision.dart [hunk 2]
```diff
@@ -152,18 +145,20 @@ class CameraMlVisionState<T> extends State<CameraMlVision<T>>
...
   }
 
-  Future<void> stopVideoRecording() async {
-    await _cameraController.stopVideoRecording();
+  Future<void> stopVideoRecording(String path) async {
+    final file = await _cameraController.stopVideoRecording();
+    await file.saveTo(path);
```
Comment (kevin-sakemaer) [inline lib/flutter_camera_ml_vision.dart]: maybe we should let the user take care of this by returning file ?

File: lib/flutter_camera_ml_vision.dart [hunk 3]
```diff
@@ -172,7 +167,8 @@ class CameraMlVisionState<T> extends State<CameraMlVision<T>>
   Future<void> takePicture(String path) async {
     await _stop(false);
     await _cameraController.initialize();
-    await _cameraController.takePicture(path);
+    final image = await _cameraController.takePicture();
+    await image.saveTo(path);
```
Comment (kevin-sakemaer) [inline lib/flutter_camera_ml_vision.dart]: same here

File: lib/flutter_camera_ml_vision.dart (hunk 2 above)
Comment (jaumard) [reply lib/flutter_camera_ml_vision.dart]: like it I'll change that!

File: lib/flutter_camera_ml_vision.dart (hunk 3 above)
Comment (jaumard) [reply lib/flutter_camera_ml_vision.dart]: Agreed

File: lib/flutter_camera_ml_vision.dart (hunk 1 above)
Comment (jaumard) [reply lib/flutter_camera_ml_vision.dart]: I don't understand the comment here, _cameraController.startVideoRecording return a future so look fine to me

Comment (kevin-sakemaer) [reply lib/flutter_camera_ml_vision.dart]: does _cameraController.startVideoRecording return `Future<void>` ?
If yes we can resolve this comment without action :)

Comment (jaumard) [reply lib/flutter_camera_ml_vision.dart]: Yep it does :)

**Review by kevin-sakemaer — APPROVED** [review]
LGTM

**Pull request merged**